source/DataBase/QueryOperationBuilder.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <format>
+#include <functional>
+#include <map>
+#include <sstream>
+#include <string>
+#include <unordered_map>
+
+#include "common/DataBaseCommon.hpp"
+#include "common/DataBaseRequest.hpp"
+#include "interface/IQueryOperationBuilder.hpp"
+
+namespace dataBase {
+
+class QueryOperationBuilder : public interface::IQueryOperationBuilder {
+  using QueryBuilders = std::unordered_map<
+      common::DataBaseRequest::OperationType,
+      std::function<std::string(const common::DataBaseRequest &,
+                                const std::string &)>>;
+
+ public:
+  QueryOperationBuilder();
+  ~QueryOperationBuilder() = default;
+  std::string buildQuery(const common::DataBaseRequest &requestData) override;
+
+ private:
+  std::string buildInsertQuery(const common::DataBaseRequest &requestData,
+                               const std::string &tableName);
+  std::string buildUpdateQuery(const common::DataBaseRequest &requestData,
+                               const std::string &tableName);
+  std::string buildDeleteQuery(const common::DataBaseRequest &requestData,
+                               const std::string &tableName);
+  std::string buildSelectQuery(const common::DataBaseRequest &requestData,
+                               const std::string &tableName);
```

Comment:
Do these methods have to be part of the interface? They are not using any class member variables. They can be moved to the unnamed namespace in the .cpp file. This way, they are only visible within the .cpp file, which further encapsulates the implementation details.

This could also improve the `QueryOperationBuilder` function to be much cleaner:
```cpp
QueryOperationBuilder::QueryOperationBuilder() {
  queryBuilders_ = {
    {common::DataBaseRequest::OperationType::Insert, buildInsertQuery},
    {common::DataBaseRequest::OperationType::Delete, buildDeleteQuery},
    {common::DataBaseRequest::OperationType::Update, buildUpdateQuery},
    {common::DataBaseRequest::OperationType::Select, buildSelectQuery}
  };
}
```